osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -259,7 +259,17 @@ public void Apply([NotNull] HitObject hitObject, [CanBeNull] HitObjectLifetimeEn
 
             // If not loaded, the state update happens in LoadComplete(). Otherwise, the update is scheduled to allow for lifetime updates.
             if (IsLoaded)
-                Schedule(() => updateState(ArmedState.Idle, true));
+            {
+                Scheduler.Add(() =>
+                {
+                    if (Result.IsHit)
+                        updateState(ArmedState.Hit, true);
+                    else if (Result.HasResult)
+                        updateState(ArmedState.Miss, true);
+                    else
+                        updateState(ArmedState.Idle, true);
```

Comment:
wondering whether a null-check here or a cancel of this schedule wouldn't be a good idea - in a theoretical scenario in which the DHO is freed before the schedule executes (e.g. if the DHO never becomes alive) this might fire at some wrong time and die on a nullref, but i'm not sure how theoretical that is

Reply:
Being a part of the hierarchy in a freed state? That sounds like something that should be asserted instead...